### ISSUE TITLE
[2.5.x] multiple 5 seconds freeze/delays while running unittests

### ISSUE THREAD

**Issue by tarmath** [playframework/playframework#7169]
Just migrated to play 2.5 from 2.4 and noticed a drastic slowdown in running unittests. 

Turns out the execution seems to wait very often for roughly 5 seconds at a time.

Also turns out this happens even in the basic play-scala template as well, so to reproduce, simply:

`activator new` then pick `play-scala` and run the tests with `activator test`

    [derpy] $ test
    [info] ApplicationSpec:
    [info] Routes
    [info] application - ApplicationTimer demo: Starting application at 2017-03-30T20:54:46.784Z.
    [info] application - ApplicationTimer demo: Stopping application at 2017-03-30T20:54:47.400Z after 1s.
    [info] - should send 404 on a bad request
    [info] HomeController
    [info] application - ApplicationTimer demo: Starting application at 2017-03-30T20:54:52.503Z.
    [DEBUG] [03/30/2017 13:54:52.562] [application-akka.actor.default-dispatcher-5] [EventStream] shutting down: StandardOutLogger started
    [info] - should render the index page
    [info] CountController
    [info] application - ApplicationTimer demo: Stopping application at 2017-03-30T20:54:52.567Z after 0s.
    [info] application - ApplicationTimer demo: Starting application at 2017-03-30T20:54:57.677Z.
    [info] application - ApplicationTimer demo: Stopping application at 2017-03-30T20:54:57.706Z after 0s.
    [info] - should return an increasing count
    [info] IntegrationSpec:
    [info] Application
    [info] application - ApplicationTimer demo: Starting application at 2017-03-30T20:55:03.583Z.
    [info] application - ApplicationTimer demo: Stopping application at 2017-03-30T20:55:04.623Z after 1s.
    [info] - should work from within a browser
    [info] ScalaTest
    [info] Run completed in 34 seconds, 868 milliseconds.
    [info] Total number of tests run: 4
    [info] Suites: completed 2, aborted 0
    [info] Tests: succeeded 4, failed 0, canceled 0, ignored 0, pending 0
    [info] All tests passed.
    [info] Passed: Total 4, Failed 0, Errors 0, Passed 4
    [success] Total time: 41 s, completed Mar 30, 2017 1:55:05 PM

Notice that there are 5 seconds delay every between every stop/start as per the logged times...

**Comment by marcospereira** [comment]
Hey @tarmath, thanks for reporting this.

Apparently, this is a problem with the logback version we are using (or how we are using it). Upgrading to the latest 1.1.x version fixed it for a project created using `sbt new`. Add the following dependencies and see if it fixes the problem for your application:

```scala
libraryDependencies += "ch.qos.logback" % "logback-core" % "1.1.11"
libraryDependencies += "ch.qos.logback" % "logback-classic" % "1.1.11"
```

If it works for your application, I would appreciate if you can submit a pull request against 2.5.x branch to update the log libraries here:

https://github.com/playframework/playframework/blob/2.5.x/framework/project/Dependencies.scala#L31-L32

I'm not sure Logback 1.2.2 (the latest Logback version) is binary compatible with 1.1.x, but I would be conservative here and update to 1.1.11.

**Comment by tarmath** [comment]
that totally did the trick for the play-scala template app:

	[derpy] $ test
	[info] Compiling 1 Scala source to /Users/mpichette/src/src/derpy/target/scala-2.11/test-classes...
	[info] IntegrationSpec:
	[info] Application
	[info] application - ApplicationTimer demo: Starting application at 2017-04-03T01:46:40.257Z.
	[info] application - ApplicationTimer demo: Stopping application at 2017-04-03T01:46:42.010Z after 2s.
	[info] - should work from within a browser
	[info] ApplicationSpec:
	[info] Routes
	[info] application - ApplicationTimer demo: Starting application at 2017-04-03T01:46:42.144Z.
	[info] application - ApplicationTimer demo: Stopping application at 2017-04-03T01:46:42.240Z after 0s.
	[info] - should send 404 on a bad request
	[info] HomeController
	[info] application - ApplicationTimer demo: Starting application at 2017-04-03T01:46:42.367Z.
	[info] application - ApplicationTimer demo: Stopping application at 2017-04-03T01:46:42.398Z after 0s.
	[info] - should render the index page
	[info] CountController
	[info] application - ApplicationTimer demo: Starting application at 2017-04-03T01:46:42.516Z.
	[info] application - ApplicationTimer demo: Stopping application at 2017-04-03T01:46:42.561Z after 0s.
	[info] - should return an increasing count
	[info] ScalaTest
	[info] Run completed in 10 seconds, 412 milliseconds.
	[info] Total number of tests run: 4
	[info] Suites: completed 2, aborted 0
	[info] Tests: succeeded 4, failed 0, canceled 0, ignored 0, pending 0
	[info] All tests passed.
	[info] Passed: Total 4, Failed 0, Errors 0, Passed 4
	[success] Total time: 16 s, completed Apr 2, 2017 6:46:43 PM

As you can see total time went from 41s to 16s... quite the difference!

**Comment by tarmath** [comment]
As for my actual application, total time has improved a bit (from 194s to 164s), but I still see several 5 seconds delays... As a hint, it seems like all tests that return error responses are delayed by 5 seconds before the response gets logged by a custom ActionBuilder.

I will try to isolate this case to reproduce it in the play-scala app and provide the code here as well, hopefully sometime tomorrow...

**Comment by tarmath** [comment]
Let me know if you still want the patch for the improvement in the meantime! 

Thanks for you help!

**Comment by marcospereira** [comment]
> I will try to isolate this case to reproduce it in the play-scala app and provide the code here as well, hopefully sometime tomorrow...

That would be very useful.

> Let me know if you still want the patch for the improvement in the meantime!

Yes! Thanks for considering this, @tarmath. :-)

**Comment by gmethvin** [comment]
Fixed by #7182

**Comment by tarmath** [comment]
this is actually still partially going on, but I haven't had the time to provide another example for it.
pretty busy at this time, but hopefully sometime next week, but I can open a new issue at that point.
